app/src/main/java/org/mozilla/fenix/settings/logins/fragment/LoginDetailFragment.kt

```diff
@@ -219,7 +219,15 @@ class LoginDetailFragment : SecureFragment(R.layout.fragment_login_detail) {
     ) : View.OnClickListener {
         override fun onClick(view: View) {
             val clipboard = view.context.components.clipboardHandler
-            clipboard.text = value
+            when (snackbarText) {
```

Comment:
Changing the behavior of this function based on the `snackbarText` that is passed in makes me think that we should probably revisit this API.

Instead of having a custom `View.OnClickListener` [I think we should follow the pattern used in other parts of the app](https://github.com/mozilla-mobile/fenix/blob/main/docs/architecture-overview.md#controller) and move this logic to a controller and call the appropriate method on the interactor
```
binding.copyUsername.setOnClickListener { interactor.copyUsername() }
```

This way we can also write tests for it as well 🙌